src/SFA.DAS.ApprenticeAccounts.Web/Pages/NotificationSettings.cshtml.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using SFA.DAS.ApprenticeAccounts.Web.Services;
+using SFA.DAS.ApprenticeAccounts.Web.Services.InnerApi;
+using SFA.DAS.ApprenticePortal.Authentication;
+using SFA.DAS.ApprenticePortal.SharedUi.Menu;
+using SFA.DAS.ApprenticePortal.SharedUi.Services;
+
+namespace SFA.DAS.ApprenticeAccounts.Web.Pages
+{
+    [HideNavigationBar]
+    public class NotificationSettingsModel : PageModel
+    {
+        private readonly ApprenticeApi _apprenticeApi;
+        private readonly NavigationUrlHelper _urlHelper;
+
+        public NotificationSettingsModel(ApprenticeApi apprenticeApi, NavigationUrlHelper urlHelper)
+        {
+            _apprenticeApi = apprenticeApi;
+            _urlHelper = urlHelper;
+        }
+
+        [BindProperty] public List<ApprenticePreference> ApprenticePreferences { get; set; }
+        [BindProperty] public string BackLink { get; set; }
+        [BindProperty] public bool SubmittedPreferences { get; set; }
+        
+
+        public async Task<IActionResult> OnGetAsync(
+            [FromServices] AuthenticatedUser user)
+        {
+
+            SubmittedPreferences = false;
+
+            try
+            {
+                var preferencesDto = await _apprenticeApi.GetAllPreferences();
+                var apprenticePreferencesDto =
+                    await _apprenticeApi.GetAllApprenticePreferencesForApprentice(user.ApprenticeId);
+
+                var apprenticePreferencesCombination = new List<ApprenticePreference>();
+
+                foreach (var preference in preferencesDto)
+                {
+                    var apprenticePreferenceDto =
+                        apprenticePreferencesDto.ApprenticePreferences.Find(ap =>
+                            ap.PreferenceId == preference.PreferenceId);
+
+                    if (apprenticePreferenceDto == null)
```

Comment:
you could probably shorten this code a bit if you think it's a little long by doing a Linq Select on the preferencesDto because you will need an apprentice preference for each one regardless, its whether there is a corresponding vale for the status which you can query when populating that field. Have a go and shout if you need help